docs/framework/guides/contributing/plugin-metadata.md

```diff
@@ -0,0 +1,33 @@
+* `path` &ndash; A path to the file, relative to the metadata file that exports the plugin.
+* `className` &ndash; The name of the class used to create the plugin.
+* `requires`&ndash; An array of the plugin's soft requirements and other non-explicit requirements.
+* `registeredToolbars`&ndash; An array of all toolbar names registered by the plugin. These names need to contain the full configuration path (e.g. `table.contentToolbar` and `table.tableToolbar` for `Table` plugin).
+* `uiComponents` &ndash; An array of objects, that describe UI components exported by the plugin. Each object in this array may contain:
+	* `name` &ndash; A name of the component the plugin exports.
+	* `type` &ndash; Component type: `Button`, `SplitButton` or `Dropdown`.
+	* `iconPath` &ndash; A path to the SVG icon for `Button` or `SplitButton` components.
+	* `label` &ndash; Text content for `Dropdown` components.
+	* `toolbars`&ndash; An array of toolbar names, a given UI component can be added to. Some features may be added to multiple toolbars.
+* `htmlOutput` &ndash; An array of objects, that define all possible HTML elements, which can be created by a given plugin. The main property in this object is `elements`. Other properties (e.g. `classes`, `styles`, `attributes`) only apply to items defined in the `elements` property within a given object. Wildcard character `*` is used to mark any value. Full list of all these properties includes:
+	* `elements` &ndash; HTML elements (a single one or an array of these) that are created by the plugin. The pseudo-element `$block` indicates that a given plugin applies classes, styles or attributes (defined in appropriate properties) for all block elements.
+	* `classes` &ndash; CSS class names (a single one or an array of these) that may be applied to the HTML elements defined in the `elements` property.
+	* `styles` &ndash; Inline CSS styles (a single one or an array of these) that may be applied to the HTML elements defined in the `elements` property.
+	* `attributes` &ndash; HTML attributes (a single one or an array of these) other than `class` and `styles` (covered separately), that might be applied to the HTML elements defined in the `elements` property.
+	* `implements` &ndash; A name of an element or a pseudo-element which indicates that HTML elements defined in the `elements` property may contain classes, styles or attributes that are created by other plugins. For example, `implements` equal to `$block` means, that HTML elements may contain classes, styles or attributes that are defined by another plugin, which have `elements` equal to `$block`.
+	* `_comment` &ndash; A human-readable description to explain more complicated cases, for example: the conditions when a given element, class or style can be created.
```

Comment:
I'd add a link to one of the metadata files that are provided by us at the bottom of this section.

Reply:
I did it a little differently: I added a link to the metadata file (not working yet, as metadata are not yet on `master` branch) at the top of this guide, so user can open a real sample file and check it while reading about the individual fields in this format. At the bottom of this guide I added a link and information on how user can see other metadata in our other packages.